baseapp/streaming.go

```diff
@@ -18,6 +18,8 @@ type ABCIListener interface {
 	ListenEndBlock(ctx types.Context, req abci.RequestEndBlock, res abci.ResponseEndBlock) error
 	// ListenDeliverTx updates the steaming service with the latest DeliverTx messages
 	ListenDeliverTx(ctx types.Context, req abci.RequestDeliverTx, res abci.ResponseDeliverTx) error
+	// ListenCommit updates the steaming service with the latest Commit event
+	ListenCommit(ctx types.Context, res abci.ResponseCommit) error
```

Comment:
👌

Reply:
Since don't backport, I'll do more interface cleanup to remove the listening stuff from caching layer.

Reply:
Broken in what way?

But yeah I'm not against it per se. Happy to backport it with @tac0turtle's blessing

Reply:
> Broken in what way?

the streaming service will associate the writes of block N with block N+1's begin blocker event, because the msg is delivered in block `N`'s commit event.

Reply:
I believe this is an internal interface. We can double check and if so, then it's safe. Currently it's hard to gauge what's internal and what's external.

Reply:
`ABCIListener` is defined in a non-internal package and is an exported identifier, so it's external.

Reply:
1000%, the statement was more like this wont effect applications. the sdk is a mess of internal things made external.

Reply:
If we assume there's no exiting use cases for `StreamingService`, then we can consider these as internal changes.